implementations/rust/ockam/ockam_identity/src/secure_channel/encryptor_worker.rs

```diff
@@ -66,9 +67,34 @@ impl EncryptorWorker {
     }
 
     /// Encrypt the message
-    async fn encrypt(&mut self, ctx: &Context, msg: SecureChannelMessage) -> Result<Vec<u8>> {
-        match self.encryptor.encrypt(&minicbor::to_vec(&msg)?).await {
-            Ok(encrypted_payload) => Ok(encrypted_payload),
+    async fn encrypt(&mut self, ctx: &Context, msg: SecureChannelMessage<'_>) -> Result<Vec<u8>> {
+        let payload = minicbor::to_vec(&msg)?;
+        // capacity is nonce + tag + payload
+        let mut buffer = Vec::with_capacity(8 + 8 + payload.len());
+        match self.encryptor.encrypt(&mut buffer, &payload).await {
+            Ok(()) => Ok(buffer),
+            // If encryption failed, that means we have some internal error,
+            // and we may be in an invalid state, it's better to stop the Worker
+            Err(err) => {
+                let address = self.addresses.encryptor.clone();
+                error!("Error while encrypting: {err} at: {address}");
+                ctx.stop_worker(address).await?;
+                Err(err)
+            }
+        }
+    }
+
+    async fn encrypt_to(
+        &mut self,
+        ctx: &Context,
+        destination: &mut Vec<u8>,
+        payload: &[u8],
+    ) -> Result<()> {
+        // by reserving the capacity beforehand, we can avoid copying memory later
+        destination.reserve(8 + 8 + 8 + payload.len());
```

Comment:
Didn't we already reserve before calling that function?

Reply:
Yes! But another user might not reserve the needed memory, this should be a no-op in most cases.